ovs/lib/scheduledtask.py

```diff
@@ -195,186 +199,220 @@ def make_timestamp(offset):
-                    ScheduledTaskController._logger.info('Execute Scrub - Virtual disk {0} - {1} - Retrieve and apply scrub work'.format(vdisk.guid, vdisk.name))
-                    work_units = locked_client.get_scrubbing_workunits()
-                    for work_unit in work_units:
-                        scrubbing_result = locked_client.scrub(work_unit, scrub_location, log_sinks=[sink])
-                        locked_client.apply_scrubbing_result(scrubbing_result)
-                    if work_units:
-                        ScheduledTaskController._logger.info('Execute Scrub - Virtual disk {0} - {1} - Scrub successfully applied'.format(vdisk.guid, vdisk.name))
+                with file_mutex(name='ovs_albaproxy_scrub', wait=lock_time):
+                    ScheduledTaskController._logger.info('Scrubber - vPool {0} - StorageRouter {1} - Deploying ALBA proxy {2}'.format(vpool.name, storagerouter.name, alba_proxy_service))
+                    client = SSHClient(storagerouter, 'root')
+                    client.dir_create(scrub_directory)
+                    client.dir_chmod(scrub_directory, 0777)  # Celery task executed by 'ovs' user and should be able to write in it
+                    if ServiceManager.has_service(name=alba_proxy_service, client=client) is True and ServiceManager.get_service_status(name=alba_proxy_service, client=client) is True:
+                        ScheduledTaskController._logger.info('Scrubber - vPool {0} - StorageRouter {1} - Re-using existing proxy service {2}'.format(vpool.name, storagerouter.name, alba_proxy_service))
+                        scrub_config = Configuration.get(scrub_config_key)
                     else:
-                        ScheduledTaskController._logger.info('Execute Scrub - Virtual disk {0} - {1} - No scrubbing required'.format(vdisk.guid, vdisk.name))
-            except Exception as ex:
-                failures.append('Failed scrubbing work unit for volume {0} with guid {1}: {2}'.format(vdisk.name, vdisk.guid, ex))
-
-        failed = len(failures)
-        ScheduledTaskController._logger.info('Execute Scrub - Finished - Success: {0} - Failed: {1} - Skipped: {2}'.format((total - failed - skipped), failed, skipped))
-        if failed > 0:
-            raise Exception('\n - '.join(failures))
-        return vdisk_guids
+                        machine_id = System.get_my_machine_id(client)
+                        port_range = Configuration.get('/ovs/framework/hosts/{0}/ports|storagedriver'.format(machine_id))
+                        port = System.get_free_ports(selected_range=port_range, nr=1, client=client)[0]
+                        # Scrub config
+                        # {u'albamgr_cfg_url': u'arakoon://config/ovs/vpools/71e2f717-f270-4a41-bbb0-d4c8c084d43e/proxies/64759516-3471-4321-b912-fb424568fc5b/config/abm?ini=%2Fopt%2FOpenvStorage%2Fconfig%2Farakoon_cacc.ini',
+                        #  u'fragment_cache': [u'none'],
```

Comment:
After discussion with @khenderick , we decided to pour this comment into another ticket
See https://github.com/openvstorage/framework/issues/964